unit_tests/test_zaza_model.py

```diff
@@ -726,6 +726,30 @@ def test_block_until_file_has_contents(self):
         self.unit2.run.assert_called_once_with(
             'cat /tmp/src/myfile.txt')
 
+    def test_block_until_file_has_contents_no_stdout(self):
```

Comment:
I'm not sure I understand why we'd have an extra function with `_no_stdout`.  Why not just handle the "no stdout" condition in the existing `test_block_until_file_has_contents`?

Reply:
I suppose to my mind we are testing two distinct things:

1. The file has contents and so the stdout key is present in the action data
2. The file has no contents and the stdout key is missing from the action data

Or are you suggesting that both scenarios should be tested under a single test method and mocks be manually reset between the two cases?

Reply:
Ah, well that's my confusion then; the 2nd should be called `test_block_until_file_has_no_contents`.  The `stdout` is a red-herring and an artifact of juju 2.7; or I'm still misunderstanding something?

Reply:
Thanks, I think you are right that renaming the test makes sense